lib/page_streaming.js

```diff
@@ -149,9 +152,52 @@ exports.PageDescriptor = PageDescriptor;
 
 /**
  * @private
+ * Creates a new object Stream which emits the resource on 'data' event.
+ * @param {ApiCall} apiCall - the callable object.
+ * @param {Object} request - the request object.
+ * @param {CallOptions=} options - the call options to customize the api call.
+ * @return {Stream} - a new object Stream.
+ */
+PageDescriptor.prototype.createStream = function(
+    apiCall, request, options) {
+  var stream = new PagedStream();
+  options = options || {};
+  options.autoPaginate = false;
+  function callback(response) {
+    var resources = response[this.resourceField];
+    for (var i = 0; i < resources.length; ++i) {
+      if (!stream.readable) {
+        return;
+      }
+      if (resources[i] === null) {
+        continue;
+      }
+      stream.push(resources[i]);
+    }
+    if (!stream.readable) {
+      return;
+    }
+    var nextPageToken = response[this.responsePageTokenField];
+    if (!nextPageToken) {
+      stream.end();
+      return;
+    }
+    options.pageToken = nextPageToken;
+    return apiCall(request, options)
+      .then(callback.bind(this))
+      .catch(stream.emit.bind(stream, 'error'));
+  }
+  apiCall(request, options)
```

Comment:
This is actually okay now, because the `apiCall(...)` now returns a Promise (previously it was an internal form of function which invokes immediately).

The function for `then` or `catch` of a Promise are actually schedule to the message loop, not invoked immediately (see the spec http://www.ecma-international.org/ecma-262/6.0/#sec-promise.prototype.then if you're interested in).